From 883a37f06da18603b8d042e630c6855b855300bb Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:52 -0400
Subject: [fs] btrfs: set right total device count for seeding support

Message-id: <1407179109-13714-346-git-send-email-sandeen@redhat.com>
Patchwork-id: 87238
O-Subject: [RHEL7.1 PATCH 345/422] Btrfs: set right total device count for seeding support
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 298658414a2f0bea1f05a81876a45c1cd96aa2e0
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Tue May 13 17:05:06 2014 +0800

    Btrfs: set right total device count for seeding support

    Seeding device support allows us to create a new filesystem
    based on existed filesystem.

    However newly created filesystem's @total_devices should include seed
    devices. This patch fix the following problem:

     # mkfs.btrfs -f /dev/sdb
     # btrfstune -S 1 /dev/sdb
     # mount /dev/sdb /mnt
     # btrfs device add -f /dev/sdc /mnt --->fs_devices->total_devices = 1
     # umount /mnt
     # mount /dev/sdc /mnt               --->fs_devices->total_devices = 2

    This is because we record right @total_devices in superblock, but
    @fs_devices->total_devices is reset to be 0 in btrfs_prepare_sprout().

    Fix this problem by not resetting @fs_devices->total_devices.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/volumes.c b/fs/btrfs/volumes.c
index 8f22a29..ab4bd39 100644
--- a/fs/btrfs/volumes.c
+++ b/fs/btrfs/volumes.c
@@ -1934,7 +1934,6 @@ static int btrfs_prepare_sprout(struct btrfs_root *root)
  fs_devices->seeding = 0;
  fs_devices->num_devices = 0;
  fs_devices->open_devices = 0;
- fs_devices->total_devices = 0;
  fs_devices->seed = seed_devices;
 
  generate_random_uuid(fs_devices->fsid);
-- 
1.7.1